sdk/core/Azure.Core.TestFramework/src/AutoRerecordingTestAttribute.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+using NUnit.Framework.Internal.Commands;
+
+namespace Azure.Core.TestFramework
+{
+    [AttributeUsage(AttributeTargets.Method)]
+    public class AutoRerecordingTestAttribute : TestAttribute, IWrapSetUpTearDown
```

Comment:
Should we just call it `RecordedTest`?

Reply:
I struggled with the naming - `RecordTest` would be less wordy and I could just add some more descriptive doc comments.

Reply:
Can't we reuse functionality of `ClientTestFixtureAttribute` to avoid putting new attribute on every method?

Reply:
I looked deeply at that question with the intent of doing this, however it's not possible with nunit. It requires that a test method, and only a test method, be decorated with an attribute implementing `IWrapSetUpTearDown` for this to work.

Also, this would be a replacement for `[Test]` as opposed to an additional attribute.

Reply:
[Here ](https://github.com/nunit/nunit/blob/c26226e4fe1a72c9851708359b01eda929fb5abb/src/NUnitFramework/framework/Internal/Execution/SimpleWorkItem.cs#L155)is how it is implemented in nunit, if you are curious.

Reply:
Well, technically it is possible to provide custom implementation of [`IMethodInfo`](https://github.com/nunit/nunit/blob/master/src/NUnitFramework/framework/Interfaces/IMethodInfo.cs) to inject custom commands into `GetCustomAttributes`, but I don't know if it worth it.

Reply:
yeah, I looked into that but decided it would not be worth it and would obfuscate what is happening.